GitHubContents.ps1

```diff
@@ -143,3 +153,266 @@ function Get-GitHubContent
+                name = $AuthorName
+                email = $AuthorEmail
+            }
+        }
+        else
+        {
+            throw 'Both AuthorName and AuthorEmail need to be specified.'
+        }
+    }
+
+    if ($PSCmdlet.ShouldProcess("$BranchName branch of $RepositoryName",
+        "Set GitHub Contents on $Path"))
+    {
+        Write-InvocationLog
+
+        $params = @{
+            UriFragment = $uriFragment
+            Description = "Writing content for $Path in the $BranchName branch of $RepositoryName"
+            Body = (ConvertTo-Json -InputObject $hashBody)
+            Method = 'Put'
+            AccessToken = $AccessToken
+            TelemetryEventName = $MyInvocation.MyCommand.Name
+            TelemetryProperties = $telemetryProperties
+            NoStatus = (Resolve-ParameterWithDefaultConfigurationValue -Name NoStatus `
+                -ConfigValueName DefaultNoStatus)
+        }
+
+        try
+        {
+            return Invoke-GHRestMethod @params
+        }
+        catch
+        {
+            $overwriteShaRequired = $false
+
+            # Temporary code to handle current differences in exception object between PS5 and PS7
+            if ($PSVersionTable.PSedition -eq 'Core')
+            {
+                $errorMessage = ($_.ErrorDetails.Message | ConvertFrom-Json).message -replace '\n',' ' -replace '\"','"'
+                if (($_.Exception -is [Microsoft.PowerShell.Commands.HttpResponseException]) -and
+                    ($errorMessage -eq 'Invalid request.  "sha" wasn''t supplied.'))
+                {
+                    $overwriteShaRequired = $true
+                }
+                else
+                {
+                    throw $_
+                }
+            }
+            else
+            {
+                $errorMessage = $_.Exception.Message  -replace '\n',' ' -replace '\"','"'
+                if ($errorMessage -like '*Invalid request.  "sha" wasn''t supplied.*')
+                {
+                    $overwriteShaRequired = $true
+                }
+                else
+                {
+                    throw $_
+                }
+            }
+
+            if ($overwriteShaRequired)
+            {
+                # Get SHA from current file
+                $getGitHubContentParms = @{
+                    Path = $Path
+                    OwnerName = $OwnerName
+                    RepositoryName = $RepositoryName
+                }
+
+                if ($PSBoundParameters.ContainsKey('BranchName'))
+                {
+                    $getGitHubContentParms['BranchName'] = $BranchName
+                }
+
+                if ($PSBoundParameters.ContainsKey('AccessToken'))
+                {
+                    $getGitHubContentParms['AccessToken'] = $AccessToken
+                }
+
+                if ($PSBoundParameters.ContainsKey('NoStatus'))
+                {
+                    $getGitHubContentParms['NoStatus'] = $NoStatus
```

Comment:
This won't work right for anyone with `DefaultNoStatus:$true` (like Pester tests).  You'll need to use `Resolve-ParameterWithDefaultConfigurationValue -Name NoStatus -ConfigValueName DefaultNoStatus`